### PR TITLE
docs: Update core getting started docs for clarity

### PR DESCRIPTION
**Pull request by travisn**
**Description of your changes:**
- Update the getting started guide, prereqs, architecture, and other basic docs with more concise wording and removed some obsolete information. Some links are removed so they don't distract the user from reading the critical information that we are conveying in the rook docs.

These updates are a result of a good conversation at Cephalocon with @zdover23 (Ceph docs lead) on what we can do to clarify and improve Rook documentation. More updates to come...

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: Documentation/Getting-Started/Prerequisites/prerequisites.md [hunk 1]
```diff
@@ -7,22 +7,22 @@ and Rook is granted the required privileges (see below for more information).
...
-In order to configure the Ceph storage cluster, at least one of these local storage types is required:
+To configure the Ceph storage cluster, at least one of these local storage types is required:
 
 * Raw devices (no partitions or formatted filesystems)
 * Raw partitions (no formatted filesystem)
 * LVM Logical Volumes (no formatted filesystem)
 * Persistent Volumes available from a storage class in `block` mode
 
-You can confirm whether your partitions or devices are formatted with filesystems with the following command:
+Confirm whether your partitions or devices are formatted with filesystems with the following command:
```
Comment (sp98) [inline Documentation/Getting-Started/Prerequisites/prerequisites.md:25]:
```suggestion
Confirm whether the partitions or devices are formatted with filesystems with the following command:
```

File: Documentation/Getting-Started/Prerequisites/prerequisites.md [hunk 2]
```diff
@@ -43,20 +43,20 @@ Enabling the Rook admission controller is recommended to provide an additional l
...
 * If encryption is enabled (`encryptedDevice: "true"` in the cluster CR)
 * A `metadata` device is specified
 
 LVM is not required for OSDs in these scenarios:
 
-* Creating OSDs on PVCs using the `storageClassDeviceSets`
+* OSDs are created on raw devices or partitions
+* OSDs are created on PVCs using the `storageClassDeviceSets`
 
 If LVM is required for your scenario, LVM needs to be available on the hosts where OSDs will be running.
```
Comment (sp98) [inline Documentation/Getting-Started/Prerequisites/prerequisites.md:61]:
```suggestion
If LVM is required, LVM needs to be available on the hosts where OSDs will be running.
```

File: Documentation/Getting-Started/Prerequisites/prerequisites.md [hunk 3]
```diff
@@ -43,20 +43,20 @@ Enabling the Rook admission controller is recommended to provide an additional l
...
 Some Linux distributions do not ship with the `lvm2` package. This package is required on all storage nodes in your k8s cluster to run Ceph OSDs.
```
Comment (sp98) [inline Documentation/Getting-Started/Prerequisites/prerequisites.md:62]:
```suggestion
Some Linux distributions do not ship with the `lvm2` package. This package is required on all storage nodes in the k8s cluster to run Ceph OSDs.
```

File: Documentation/Getting-Started/example-configurations.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 title: Example Configurations
 ---
 
-Configuration for Rook and Ceph can be configured in multiple ways to provide block devices, shared filesystem volumes or object storage in a kubernetes namespace. We have provided several examples to simplify storage setup, but remember there are many tunables and you will need to decide what settings work for your use case and environment.
+Configuration for Rook and Ceph can be configured in multiple ways to provide block devices, shared filesystem volumes or object storage in a kubernetes namespace. While several examples are provided to simplify storage setup, there are various settings that you will need to configure to optimize for your production environment.
```
Comment (sp98) [inline Documentation/Getting-Started/example-configurations.md:5]:
```suggestion
Configuration for Rook and Ceph can be configured in multiple ways to provide block devices, shared filesystem volumes or object storage in a kubernetes namespace. While several examples are provided to simplify storage setup, various settings will be needed to configure in order to optimize the production environment.
```

**Review by sp98 — CHANGES_REQUESTED** [review]
Few nits about `you` and `your` references. You can ignore them if they are not very relevant.

**Comment by sp98** [comment]
the docs-check CI seems to be complaining about the symlink location.

**Comment by travisn** [comment]
> Few nits about `you` and `your` references. You can ignore them if they are not very relevant.

Great suggestions, I was getting tired staring at the docs for so long. This is going to take many separate PRs. :)

File: Documentation/Getting-Started/quickstart.md [hunk 5]
```diff
@@ -41,11 +40,20 @@ kubectl create -f crds.yaml -f common.yaml -f operator.yaml
...
-After the cluster is running, you can create [block, object, or file](#storage) storage to be consumed by other applications in your cluster.
+After the cluster is running, applications can consume [block, object, or file](#storage) storage.
 
 ## Deploy the Rook Operator
 
-The first step is to deploy the Rook operator. Check that you are using the [example yaml files](https://github.com/rook/rook/blob/master/deploy/examples) that correspond to your release of Rook. For more options, see the [example configurations documentation](example-configurations.md).
+The first step is to deploy the Rook operator.
+
+!!! important
+  The [Rook Helm Chart](../Helm-Charts/operator-chart.md) is available to deploy the operator instead of creating the below manifests.
```
Comment (galexrt) [inline Documentation/Getting-Started/quickstart.md:50]: Please check locally using `make docs-preview` if two spaces was enough, these "hint blocks" might require four spaces

**Comment by travisn** [comment]
> the docs-check CI seems to be complaining about the symlink location.

For now I'm going to revert those changes that symlink these docs, and consider how to structure the troubleshooting docs in a separate PR.

**Review by galexrt — APPROVED** [review]

**Review by sp98 — APPROVED** [review]
